torch/serialization.py

```diff
@@ -760,7 +781,17 @@ def load(
                                   " silence this warning)", UserWarning)
                     opened_file.seek(orig_position)
                     return torch.jit.load(opened_file, map_location=map_location)
+                if weights_only:
+                    try:
+                        return _load(opened_zipfile, map_location, _weights_only_unpickler, **pickle_load_args)
+                    except RuntimeError as e:
+                        raise pickle.UnpicklingError(UNSAFE_MESSAGE + str(e)) from None
```

Comment:
Are we sure we definitely want to suppress the internal exception?

Reply:
All error messages raised by `_weights_only_unpickler` are unique, but on the other hand error message is much cleaner than way.